tests/test_composite.py

```diff
@@ -324,18 +324,35 @@ def test_multi_slice_index(ant, sphere, uniform, airplane, globe):
         assert id(sub[i]) == id(multi[i])
         assert sub.get_block_name(i) == multi.get_block_name(i)
     sub = multi[0:-1]
-    assert len(sub) == len(multi) == multi.n_blocks
-    for i in range(multi.n_blocks):
+    assert len(sub) + 1 == len(multi) == multi.n_blocks
+    for i in range(multi.n_blocks - 1):
         assert id(sub[i]) == id(multi[i])
         assert sub.get_block_name(i) == multi.get_block_name(i)
     sub = multi[0:-1:2]
-    assert len(sub) == 3
-    for i in range(3):
+    assert len(sub) == 2
+    for i in range(2):
         j = i*2
         assert id(sub[i]) == id(multi[j])
         assert sub.get_block_name(i) == multi.get_block_name(j)
 
 
+def test_slice_defaults(ant, sphere, uniform, airplane, globe):
+    multi = multi_from_datasets(ant, sphere, uniform, airplane, globe)
+    assert multi[:] == multi[0:len(multi)]
+
+
+def test_slice_negatives(ant, sphere, uniform, airplane, globe):
+    multi = multi_from_datasets(ant, sphere, uniform, airplane, globe)
+    reverse_multi = multi_from_datasets(globe, airplane, uniform, sphere, ant)
+    assert multi[::-1] == reverse_multi
```

Comment:
When adding key comparison to equality check, this fails because pyvista adds a key string to each block

https://github.com/pyvista/pyvista/blob/main/pyvista/core/composite.py#L468

I feel this is unexpected, but a thorough equality should check this.

Reply:
I'm on record saying that [the current `MultiBlock` API is a bit crazy to begin with](https://github.com/pyvista/pyvista/issues/1507). So some surprises are expected (heh).

I would say that in the above `multi` and `reverse_multi` are not _inherently_ reverses of each other, so this assertion failing is fine by me. What I would only expect to be a hard truth is the same for
```py
    reverse_multi = pv.MultiBlock({key: multi[key] for key in multi.keys()[::-1]})
```
(or something similar).

Reply:
Used your suggestion, which is a good one.  It makes it very clear that we are testing for the same thing.  The awkwardness here highlights the weirdness of this API for multiblock.